x/ibc/20-transfer/types/packet.go

```diff
@@ -62,43 +63,43 @@ func (pd PacketDataTransfer) ValidateBasic() error {
 	return nil
 }
 
-// Implements channelexported.PacketDataI
+// GetBytes implements channelexported.PacketDataI
 func (pd PacketDataTransfer) GetBytes() []byte {
 	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(pd))
 }
 
-// Implements channelexported.PacketDataI
+// GetTimeoutHeight implements channelexported.PacketDataI
 func (pd PacketDataTransfer) GetTimeoutHeight() uint64 {
 	return pd.Timeout
 }
 
-// Implements channelexported.PacketDataI
+// Type implements channelexported.PacketDataI
 func (pd PacketDataTransfer) Type() string {
 	return "ics20/transfer"
 }
 
 var _ channelexported.PacketDataI = AckDataTransfer{}
 
-type AckDataTransfer struct {
-}
+// AckDataTransfer is a no-op packet
+// See spec for onAcknowledgePacket: https://github.com/cosmos/ics/tree/master/spec/ics-020-fungible-token-transfer#packet-relay
+type AckDataTransfer struct{}
 
-// Implements channelexported.PacketDataI
-// ValidateBasic performs a basic check of the packet fields
+// ValidateBasic implements channelexported.PacketDataI
 func (ack AckDataTransfer) ValidateBasic() error {
 	return nil
 }
 
-// Implements channelexported.PacketDataI
+// GetBytes implements channelexported.PacketDataI
 func (ack AckDataTransfer) GetBytes() []byte {
 	return []byte("ok")
```

Comment:
Ideally, I'd want this to be something like
```go
	return []byte(fmt.Sprintf("ack %s", string(ack.PacketHash()))
```

Reply:
See above comment, if we want to handle failed transfers this data needs to be structured.

Reply:
Lets handle failed transfers